### PR TITLE
Add SearchApi documentloader, tool & podcast QA example

### PR DESCRIPTION
**Pull request by SebastjanPrachovskij**
Changes:

- Added SearchApi document loader for Web Search Results.
- Added SearchApi tool for real-time Google Search queries.
- Added "Chat with podcast" example. 

Feel free to assign @SebastjanPrachovskij as a main reviewer for any SearchApi PRs. We will be glad to help and support Flowise development.

---
Usage:

image

### CODE REVIEW

File: packages/components/credentials/SearchApi.credential.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    name: string
+    version: number
+    description: string
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'Search API'
+        this.name = 'searchApi'
+        this.version = 1.0
+        this.inputs = [
```
Comment (HenryHengZJ) [inline packages/components/credentials/SearchApi.credential.ts:14]: would be great if we can add a `description` on how to get the API key. Example [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/credentials/AWSCredential.credential.ts#L15)

File: packages/server/marketplaces/chatflows/Chat with a Podcast.json [hunk 2]
```diff
@@ -0,0 +1,702 @@
+{
```
Comment (HenryHengZJ) [inline packages/server/marketplaces/chatflows/Chat with a Podcast.json:1]: IMO this template is almost same as `Conversational Retrieval QA Chain` template. Maybe a template that uses an agent + search tool?

File: packages/components/credentials/SearchApi.credential.ts (hunk 1 above)
Comment (SebastjanPrachovskij) [reply packages/components/credentials/SearchApi.credential.ts:16]: Thanks, that's a great suggestion!

File: packages/server/marketplaces/chatflows/Chat with a Podcast.json (hunk 2 above)
Comment (SebastjanPrachovskij) [reply packages/server/marketplaces/chatflows/Chat with a Podcast.json:1]: I have based this on [LangChainJS use case example](https://github.com/langchain-ai/langchainjs/blob/main/examples/src/use_cases/youtube/chat_with_podcast.ts).

It demonstrates the usage of a Web Search document loader (YouTube transcripts in this case). `Conversational Retrieval QA Chain` is a generalized example and this chat flow is a specific use case. 

> Maybe a template that uses an agent + search tool?

`MRKLAgent` example is already occupied. Or do you have something else in mind?

I'm happy to make changes if needed, just let me know.

Comment (HenryHengZJ) [reply packages/server/marketplaces/chatflows/Chat with a Podcast.json:1]: ah okay, i'm good with it 👍

**Comment by HenryHengZJ** [comment]
if you can run `yarn lint-fix` @SebastjanPrachovskij , everything else looks good!

**Comment by SebastjanPrachovskij** [comment]
@HenryHengZJ Done, thank you!

**Pull request merged**